src/sync.rs

```diff
@@ -221,17 +223,149 @@ impl<K: Eq + Hash, V: StableDeref> FrozenMap<K, V> {
     // TODO add more
 }
 
+impl<K: Clone, V> FrozenMap<K, V> {
+    pub fn keys_cloned(&self) -> Vec<K> {
+        self.map.read().unwrap().keys().cloned().collect()
+    }
+}
+
+impl<K: Eq + Hash, V: Copy> FrozenMap<K, V> {
+    /// Returns a copy of the value corresponding to the key.
+    ///
+    /// The key may be any borrowed form of the map's key type, but
+    /// [`Hash`] and [`Eq`] on the borrowed form *must* match those for
+    /// the key type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use elsa::sync::FrozenMap;
+    ///
+    /// let map = FrozenMap::new();
+    /// map.insert(1, 6);
+    /// assert_eq!(map.get_copy(&1), Some(6));
+    /// assert_eq!(map.get_copy(&2), None);
+    /// ```
+    pub fn get_copy<Q: ?Sized>(&self, k: &Q) -> Option<V>
+    where
+        K: Borrow<Q>,
+        Q: Hash + Eq,
+    {
+        let map = self.map.read().unwrap();
+        map.get(k).cloned()
+    }
+
+    /// If the key exists in the map, returns a reference
+    /// to the corresponding value, otherwise inserts a
+    /// new entry in the map for that key and returns a
+    /// reference to the given value.
+    ///
+    /// Existing values are never overwritten.
+    ///
+    /// The key may be any borrowed form of the map's key type, but
+    /// [`Hash`] and [`Eq`] on the borrowed form *must* match those for
+    /// the key type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use elsa::sync::FrozenMap;
+    ///
+    /// let map = FrozenMap::new();
+    /// assert_eq!(map.insert(1, Box::new("a")), &"a");
+    /// assert_eq!(map.insert(1, Box::new("b")), &"a");
+    /// ```
+    pub fn get_copy_or_insert(&self, k: K, v: V) -> V {
+        let mut map = self.map.write().unwrap();
+        let inserted = map.entry(k).or_insert(v);
```

Comment:
nit: please include comments saying that this is safe since it does not overwrite existing entries (below too)

Reply:
I updated the documentation comments. Is that what you're looking for?